### DISCUSSION TITLE
How do I trigger iTerm2 style global access to ghostty?

### DISCUSSION THREAD

**Discussion by Integralist** [ghostty-org/ghostty#3141] [Q&A] [answered]
👋 

Apologies if this is really obvious but I read through the docs and couldn't find any mention of what the key combination is to trigger the "global hotkey" feature (e.g. where you can at any point in time open a floating instance of ghostty).

Thanks for any guidance.

**Comment by gpanders** [comment, marked as answer]
The keybind action you're looking for is `toggle_quick_terminal`.

**Reply by qwerasd205** [reply]
In combination with the `global:` modifier (https://ghostty.org/docs/config/keybind#global:)

**Comment by writes-of-spring** [comment]
Yeah definitely a muscle memory for me now that I'd be sad to lose out on.

```toggle_quick_terminal``` is a singleton so it's not quite the same. 

Iterm's hotkey is a show/hide all windows and focuses the most recently used tab.

```keybind = global:cmd+§=toggle_visibility``` got me close, but feels a bit quirky, like if I focus on another application, it'll first hide the windows, and it'll reopen windows I had had closed with no remaining tabs cmd cmd-w

**Reply by tholu** [reply]
I have the same problem, why would the dropdown console be a singleton and not support tabs? :(

**Reply by mitchellh** [reply]
It's noted in the docs: macOS doesn't allow it for non-titled windows. This is a macOS issue unless we implement a fully custom tab bar.

**Reply by tholu** [reply]
There also is one or multiple bugs(?) with `toggle_visibility`, when I do the following:

1. Open new Ghostty Window
2. Create a new tab
3. Use keybind for toggling visibility twice
4. The tabs vanish (why?)
5. I close extra window with `Cmd+w`
6. When cycling through Ghostty Windows with Cmd+`, I suddenly get empty windows that also don't react to `Cmd+w` anymore

![CleanShot 2024-12-26 at 23 17 36@2x](https://github.com/user-attachments/assets/75b50e81-da60-4fa0-b5b0-2de9e8703344)

(I used `keybind = global:ctrl+,=toggle_visibility` as config)

**Reply by Integralist** [reply]
yeah I have a similar issue with: 

```
keybind = global:cmd+§=toggle_visibility
```

sometimes I have to press it twice to get it to show ghostty again but often it'll open a new ghostty instance next to my existing ghostty instance and it'll be blank and doesn't actually load any shell?

<img width="1286" alt="Screenshot 2024-12-27 at 13 34 01" src="https://github.com/user-attachments/assets/5dcd9107-f5d0-4c5b-a74d-74e2b9c7a8bb" />

**Reply by Integralist** [reply]
That said, `toggle_quick_terminal` is really what I was after (not `toggle_visibility`)

**Comment by cameronmurphy** [comment]
I'm still interested in the iTerm behaviour whereby the existing Ghostty app gains focus the first time I execute the shortcut. Currently when Ghostty isn't the focused app on macOS, the first press of the shortcut hides Ghostty so to get from another app back to Ghostty I have to press the shortcut twice.
I have it mapped on macOS 15.2 (24C101) and Ghostty 1.0.1 build 8696 to Ctrl + Tilde.
```ini
keybind = global:ctrl+grave_accent=toggle_visibility
```

**Comment by mhenrixon** [comment]
I'd love to have profiles and be able to assign a hotkey to open that profile globally but I have sort of solved it now with Hammerspoon app and the following config:

```lua
hs.hotkey.bind({"cmd"}, "escape", function()
  local ghostty = hs.application.find("Ghostty")
  if ghostty and ghostty:isFrontmost() then
    ghostty:hide()
  else
    hs.application.launchOrFocus("Ghostty")
  end
end)
```

**Reply by aquarhead** [reply]
Thanks, this works great!

This almost completes the replacement of iTerm2 for me, the only other (minor) features are profiles (although I don't use a hot key profile, but I do have a separate profile for presentation - large font and no transparency) and "Show Timestamps" (sometimes useful for postmortems)

**Reply by mhenrixon** [reply]
I miss CMD click to open files at the correct line number in editor and I miss copy on select.

Those two and I'm great.

**Reply by aquarhead** [reply]
These is `copy-on-select = true` (I'm running latest version)

BTW the hs script seems sometimes fail to detect `ghostty` is `nil` for whatever reason. I found Raycast has the Hotkey feature for any app just like what iTerm2 implemented so I've switched to that :)

**Reply by tholu** [reply]
@aquarhead Can you share the exact way to set this up in Raycast? Very much appreciated.

**Reply by aquarhead** [reply]
@tholu Just search for the app name and set a keybind for "Hotkey"

![image](https://github.com/user-attachments/assets/1eb2b5c6-6467-41c9-922a-f6bd27785356)

**Comment by steipete** [comment]
To join in here, `toggle_quick_terminal` not working when Ghostty is not in foreground is my main blocker from moving over from iTerm. I do use global but that doesn't work.

`keybind = global:ctrl+grave_accent=toggle_quick_terminal`

Is there already a bug tracking this, or is it supposed to only work when selected?

**Reply by writes-of-spring** [reply]
Make sure you've updated the 1.1, I've found it to feel just like iterm's toggle!

**Reply by steipete** [reply]
Rebooting helped! I was on the latest version already tho. So ignore my message, if it becomes unreliable again I'll see if I find a pattern and report back.
